### PR TITLE
Update default TFMs without requiring a new SDK

### PR DESCRIPTION
**Pull request by ViktorHofer**
NetCurrent -> net9.0
NetMinimum -> net8.0
Introduce NetToolCurrent for msbuild tasks and other tools.
Update TFM description

**Allow repositories using the Arcade.Sdk to target net9.0 without requiring an SDK update. This works for future TFMs as well.**

I tested the changes in the windowsdesktop repository.

---

What remains is avoiding hardcoded TFMs in props/targets files. Example: https://github.com/dotnet/xliff-tasks/blob/d66a09c8b11e1235273c8e6cd946e5c8e247f189/src/XliffTasks/build/Microsoft.DotNet.XliffTasks.targets#L5-L6

Filed https://github.com/dotnet/arcade/issues/14138 to discuss a solution to that problem.

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkDefaults.props [hunk 1]
```diff
@@ -1,28 +1,110 @@
...
+      Include="@(KnownFrameworkReference->WithMetadataValue('Identity', 'Microsoft.WindowsDesktop.App')->WithMetadataValue('TargetFramework', '$(NetToolCurrent)'))"
+      TargetFramework="$(NetCurrent)"
+      Condition="@(KnownFrameworkReference->WithMetadataValue('Identity', 'Microsoft.WindowsDesktop.App')->WithMetadataValue('TargetFramework', '$(NetCurrent)')) == ''" />
+
+    <KnownFrameworkReference
+      Include="@(KnownFrameworkReference->WithMetadataValue('Identity', 'Microsoft.WindowsDesktop.App.WPF')->WithMetadataValue('TargetFramework', '$(NetToolCurrent)'))"
+      TargetFramework="$(NetCurrent)"
+      Condition="@(KnownFrameworkReference->WithMetadataValue('Identity', 'Microsoft.WindowsDesktop.App.WPF')->WithMetadataValue('TargetFramework', '$(NetCurrent)')) == ''" />
+
+    <KnownFrameworkReference
```
Comment (mmitche) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkDefaults.props:93]: Does this end up bringing in WD even if it's not used?

**Comment by mmitche** [comment]
Are you sure it's a good idea to enable this by default?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkDefaults.props (hunk 1 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/TargetFrameworkDefaults.props:93]: Nope, a `KnownFrameworkReference` is just an indication of what is known by the SDK to target. Defining an `FrameworkReference` item would make it being used.

**Comment by ViktorHofer** [comment]
Putting this back to draft as there are issues when doing this but consuming live runtime dependencies (i.e. Microsoft.Extensions packages). The net9.0 TFM would mean we would still build agains the 8.0.0.0 versioned targeting pack assemblies but the NuGet packages already reference 9.0.0.0 versioned assemblies. That ultimately results in a torn state and these compiler errors:

> CSC : error CS1705: Assembly 'Microsoft.Extensions.FileSystemGlobbing' with identity 'Microsoft.Extensions.FileSystemGl
obbing, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' uses 'System.Runtime, Version=9.0.0.0, Cultu
re=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version than referenced assembly 'System.Runtime' with
identity 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' [C:\git\format\src\dotnet-
format.csproj

**Pull request closed without merging**